### PR TITLE
Throw AuthenticationException if failed to conn due to auth errors

### PR DESCRIPTION
**Pull request by zhenlineo**
This includes the error that the client provide a wrong credentials and
the server provides a wrong certificate.

### CODE REVIEW

File: driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java [hunk 1]
```diff
@@ -81,7 +82,7 @@ public static TLSSocketChannel create( ByteChannel channel, Logger logger, SSLEn
         }
         catch ( SSLHandshakeException e )
         {
-            throw new ClientException( "Failed to establish secured connection with the server: " + e.getMessage(), e );
+            throw new UnauthorizedException( "Failed to establish secured connection with the server: " + e.getMessage(), e );
```
Comment (technige) [inline driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java]: I'm not sure that we should reuse `UnauthorizedException` for TLS handshake failure as well as for actual login failure. The Python driver uses `ServiceUnavailable` here but that doesn't seem quite right either.

Perhaps we should have something separate such as `SecurityLayerUnavailable`?

Comment (zhenlineo) [reply driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java]: I would prefer to have fewer public errors. So how about change both of them to `SecurityException`/`AuthorizationException`?

Comment (lutovich) [reply driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java]: +1 for `AuthorizationException`

**Review by lutovich — COMMENTED** [review]
@zhenlineo changes look good, name of `UnauthorizedException` is probably the only thing that could be improved.

File: driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java (hunk 1 above)
Comment (technige) [reply driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java]: I think the name is fine as-is for login failure but why would fewer errors be useful here? Login failure _could_ trigger an application code path that requires the user to enter new login details whereas an SSL handshake failure should not.

**Review by lutovich — APPROVED** [review]

**Comment by zhenlineo** [comment]
Will merge this one latter after the big routing specification change, as it should be easier to rebase this PR

**Pull request merged**